components/client/src/test/unit/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagatorTest.java

```diff
@@ -142,58 +119,32 @@ public void doesNotPropagateErrorsTwice() throws Exception {
         EmbeddedChannel channel = new EmbeddedChannel(handler);
 
         channel.writeInbound(httpResponseHeaders);
-        channel.writeInbound(newHttpContent("one"));
 
-        LiveHttpResponse response = responseSubscriber.getOnNextEvents().get(0);
-        subscribeToContent(response);
+        LiveHttpResponse response = verifyNextCalledOnResponseSubscriber();
 
-        channel.pipeline().fireExceptionCaught(new RuntimeException("Simulated exception: something went horribly wrong!"));
-        assertThat(responseSubscriber.getOnErrorEvents().size(), is(1));
+        StepVerifier.create(response.body())
+                .then(channel::runPendingTasks) // Execute onSubscribe in FSM
+                .then(() -> channel.pipeline().fireExceptionCaught(new RuntimeException())) // Will emmit BadHttpResponseException
+                .then(() -> channel.pipeline().fireChannelInactive()) // Will emmit TransportLostException
+                .expectError(BadHttpResponseException.class)
+                .verify();
 
-        channel.pipeline().fireChannelInactive();
-        assertThat(responseSubscriber.getOnErrorEvents().size(), is(1));
+        verify(responseSubscriber, atMostOnce()).error(any());
     }
 
 
     @Test
     public void handlesIdleStateEvent() throws Exception {
         EmbeddedChannel channel = new EmbeddedChannel(new NettyToStyxResponsePropagator(responseSubscriber, SOME_ORIGIN));
         channel.writeInbound(httpResponseHeaders);
-        channel.writeInbound(newHttpContent("one"));
-
-        channel.pipeline().fireUserEventTriggered(ALL_IDLE_STATE_EVENT);
-
-        List<Throwable> errors = responseSubscriber.getOnErrorEvents();
-        assertThat(errors, hasSize(1));
-        assertThat(errors.get(0), is(instanceOf(ResponseTimeoutException.class)));
-    }
-
-
-    @Test
-    public void pushesHttpContentOnChannelReadComplete() {
```

Comment:
Is the functionality for "pushesHttpContentOnChannelReadComplete" still being tested somewhere?

Reply:
Okay, please disregard my comment.